pirouette-plutusir/src/Language/Pirouette/PlutusIR/Syntax.hs

```diff
@@ -0,0 +1,405 @@
+  parseBuiltinTerm =
+    label "Builtin Term" $
+      asum $
+        map
+          try
+          [ -- Integers
+            AddInteger <$ symbol "b/addInteger",
+            SubtractInteger <$ symbol "b/subtractInteger",
+            MultiplyInteger <$ symbol "b/multiplyInteger",
+            DivideInteger <$ symbol "b/divideInteger",
+            QuotientInteger <$ symbol "b/quotientInteger",
+            RemainderInteger <$ symbol "b/remainderInteger",
+            ModInteger <$ symbol "b/modInteger",
+            EqualsInteger <$ symbol "b/equalsInteger",
+            LessThanInteger <$ symbol "b/lessThanInteger",
+            LessThanEqualsInteger <$ symbol "b/lessThanEqualsInteger",
+            -- Bytestrings
+            AppendByteString <$ symbol "b/appendByteString",
+            ConsByteString <$ symbol "b/consByteString",
+            SliceByteString <$ symbol "b/sliceByteString",
+            LengthOfByteString <$ symbol "b/lengthOfByteString",
+            IndexByteString <$ symbol "b/indexByteString",
+            EqualsByteString <$ symbol "b/equalsByteString",
+            LessThanByteString <$ symbol "b/lessThanByteString",
+            LessThanEqualsByteString <$ symbol "b/lessThanEqualsByteString",
+            -- Cryptography and hashes
+            Sha2_256 <$ symbol "b/sha2",
+            Sha3_256 <$ symbol "b/sha3",
+            Blake2b_256 <$ symbol "b/blake2b",
+            VerifySignature <$ symbol "b/verifySignature",
+            -- P.VerifyEd25519Signature <$ symbol "b/verifyEd25519Signature",
+            -- P.VerifyEcdsaSecp256k1Signature <$ symbol "b/verifyEcdsaSecp256k1Signature",
+            -- P.VerifySchnorrSecp256k1Signature <$ symbol "b/verifySchnorrSecp256k1Signature",
+            -- Strings
+            AppendString <$ symbol "b/appendString",
+            EqualsString <$ symbol "b/equalsString",
+            EncodeUtf8 <$ symbol "b/encodeUtf8",
+            DecodeUtf8 <$ symbol "b/decodeUtf8",
+            -- Tracing
+            Trace <$ symbol "b/trace"
+          ]
+
+  -- Some builtins will also be available through more familiar infix operators
+  operators =
+    [ [ InfixR (symbol "*" >> return (exprBinApp MultiplyInteger)),
+        InfixR (symbol "/" >> return (exprBinApp DivideInteger)),
+        InfixR (symbol "%" >> return (exprBinApp ModInteger))
+      ],
+      [ InfixR (symbol "+" >> return (exprBinApp AddInteger)),
+        InfixR (symbol "-" >> return (exprBinApp SubtractInteger))
+      ],
+      [ InfixN (symbol "<" >> return (exprBinApp LessThanInteger)),
+        InfixN (symbol "<=" >> return (exprBinApp LessThanEqualsInteger)),
+        InfixN (symbol "==i" >> return (exprBinApp EqualsInteger)),
+        InfixN (symbol "==bs" >> return (exprBinApp EqualsByteString)),
+        InfixN (symbol "==s" >> return (exprBinApp EqualsString))
```

Comment:
You can do `smth $> x` (coming from `Data.Functor`) instead of `smth >> pure x`, I think.